### ISSUE TITLE
Consider setting up Kibana dashboards only once

### ISSUE THREAD

**Issue by david-kow** [elastic/cloud-on-k8s#3307]
We've received some feedback (thanks @exekias!) regarding Kibana dashboard setup using `kibanaRef` in Beat CRD. The concern is that every time we redo the setup from all Beat instances. This causes at least two issues:
1. longer Beat startup time
1. unnecessary load on ES/Kibana

A solution proposed was to use Job to do it once. 

With that solution (and probably any other) I do see some issues revolving around detecting whether this was already done:
1. do we run it again on config change or version upgrade?
1. how do we know if Kibana already has those dashboards? `kibanaRef` might not change, but the instance may be recreated.
1. how do we know if Kibana is missing some of the dashboards? Should the operator try to query for them? How to know which dashboards are expected? Would we need to track them between Beat versions?

I think this might end up being a difficult problem to solve, so additional two approaches:
1. run Job once for every config change that would cause Pods to be restarted (reduce `n` setups to `1`), or
1. document to remove `kibanaRef` if user doesn't wish for setups to be run anymore

**Comment by pebrc** [comment]
I am closing this for now given that the recommended way to use Beats today is through Elastic Agent and Fleet with the Elastic Package registry taking care of dashboard installation, which does not have the problem described in this issue. We can reopen if necessary.
